library/src/meta/footnote.rs

```diff
@@ -73,14 +157,27 @@ impl Synthesize for FootnoteElem {
 impl Show for FootnoteElem {
     #[tracing::instrument(name = "FootnoteElem::show", skip_all)]
     fn show(&self, vt: &mut Vt, styles: StyleChain) -> SourceResult<Content> {
-        let loc = self.0.location().unwrap();
-        let numbering = self.numbering(styles);
-        let counter = Counter::of(Self::func());
-        let num = counter.at(vt, loc)?.display(vt, &numbering)?;
-        let sup = SuperElem::new(num).pack();
-        let hole = HElem::new(Abs::zero().into()).with_weak(true).pack();
-        let loc = self.0.location().unwrap().variant(1);
-        Ok(hole + sup.linked(Destination::Location(loc)))
+        let display = |vt: &mut Vt| {
+            let loc = self.declaration_location(vt).at(self.span())?;
+            let numbering = self.numbering(styles);
+            let counter = Counter::of(Self::func());
+            let num = counter.at(vt, loc)?.display(vt, &numbering)?;
+            let sup = SuperElem::new(num).pack();
+            let hole = HElem::new(Abs::zero().into()).with_weak(true).pack();
+            let loc = self.body_location(vt).at(self.span())?;
+            Ok(hole + sup.linked(Destination::Location(loc)))
+        };
+        if self.is_ref() {
+            Ok(vt.delayed(display))
```

Comment:
I think it would be okay to always call `.delayed` here. Shouldn't have any negative consequences.